Add a hyperlink to text docs.ansible.com for testing site

bmillemathias-1a opened the pull request:
##### SUMMARY
Add a hyperlink to text docs.ansible.com for testing site

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
docs.ansible.com

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```paste below

```

bmillemathias committed:
Add a hyperlink to text docs.ansible.com on testing site

bmillemathias-1a commented:
+docsite

acozine commented:
@bmillemathias-1a great idea! The current branch doesn't quite do the right thing when published:
![Screen Shot 2020-07-23 at 11 49 42 AM](https://user-images.githubusercontent.com/879121/88314566-b9c73b00-ccda-11ea-958f-b437c3fbb084.png)

acozine commented on docs/docsite/_themes/sphinx_rtd_theme/ansible_banner.html:
```diff
@@ -4,7 +4,7 @@
       document.write('<div id="testing_banner_id" class="admonition important">');
       para = document.createElement('p');
-      banner_text=document.createTextNode("This is a testing site. For the official docs go to https://docs.ansible.com/");
+      banner_text=document.innerHTML('<b>This is the testing site for Ansible Documentation<b/>. For the official documentation website visit <a href="https://docs.ansible.com/">https://docs.ansible.com/</a>');
```
Should the end-tag be `</b>` instead of `<b/>`?

bmillemathias replied:
indeed

acozine commented on docs/docsite/_themes/sphinx_rtd_theme/ansible_banner.html:
```diff
@@ -4,7 +4,7 @@
       document.write('<div id="testing_banner_id" class="admonition important">');
       para = document.createElement('p');
-      banner_text=document.createTextNode("This is a testing site. For the official docs go to https://docs.ansible.com/");
+      banner_text=document.innerHTML('<b>This is the testing site for Ansible Documentation<b/>. For the official documentation website visit <a href="https://docs.ansible.com/">https://docs.ansible.com/</a>');
```
```suggestion
      banner_text=document.innerHTML('<b>This is the testing site for Ansible Documentation</b>. Unless you're reviewing pre-production changes, please visit the <a href="https://docs.ansible.com/ansible/latest/">official documentation website</a>');
```

bmillemathias commented:
![image](https://user-images.githubusercontent.com/8579/88454978-c7172d80-ce72-11ea-9d02-cf7f94352082.png)

I tuned on my local copy the css to space properly the two banners

bmillemathias committed:
Remove overlapping banner

i'm pretty sure this is not the correct place because the css is
minified however that can start something

felixfontein commented on docs/docsite/_themes/sphinx_rtd_theme/ansible_banner.html:
```diff
@@ -4,7 +4,7 @@
       document.write('<div id="testing_banner_id" class="admonition important">');
       para = document.createElement('p');
-      banner_text=document.createTextNode("This is a testing site. For the official docs go to https://docs.ansible.com/");
+      para.innerHTML('<b>This is the testing site for Ansible Documentation</b>. For the official documentation website visit the <a href="https://docs.ansible.com/latest/">official documentation website</a>.');
```
```suggestion
      para.innerHTML = '<b>This is the testing site for Ansible Documentation</b>. For the official documentation website visit the <a href="https://docs.ansible.com/latest/">official documentation website</a>.';
```

bmillemathias committed:
Update docs/docsite/_themes/sphinx_rtd_theme/ansible_banner.html

Co-authored-by: Felix Fontein <felix@fontein.de>

bmillemathias commented:
/rebuild_failed

samccann commented:
Thanks for the hard work here @bmillemathias-1a !  I had to add another banner update, so I absorbed this work into https://github.com/ansible/ansible/pull/71360 and will close this one.

bmillemathias commented:
> 
> 
> Thanks for the hard work here @bmillemathias-1a ! I had to add another banner update, so I absorbed this work into #71360 and will close this one.

hard work, hum hum :)

felixfontein commented:
Sorry for only spotting that now, but the banner links to https://docs.ansible.com/latest/ which is 404. I guess it should be https://docs.ansible.com/ansible/latest/ ?

bmillemathias commented:
> 
> 
> Sorry for only spotting that now, but the banner links to https://docs.ansible.com/latest/ which is 404. I guess it should be https://docs.ansible.com/ansible/latest/ ?

I saw in #71360 @samccann put the correct URL.
